model/config.go

```diff
@@ -61,34 +44,68 @@ func (k Key) String() string {
 // Object references supplied and returned from this interface should be
 // treated as read-only. Modifying them violates thread-safety.
 type ConfigRegistry interface {
-	// Get retrieves a configuration element, bool indicates existence
-	Get(key Key) (proto.Message, bool)
+	// Get retrieves a configuration element.
+	Get(kind, key string) (config proto.Message, exists bool, revision Revision)
 
-	// List returns objects for a kind in a namespace.
-	// Use namespace "" to list resources across namespaces
-	List(kind string, namespace string) (map[Key]proto.Message, error)
+	// List returns objects by type indexed by the key
+	List(kind string) (map[string]proto.Message, error)
```

Comment:
Listing is typically 1 level deep. based on the resource scheme `/route-rule/{destination-service}/{rule-name}`    List("route-rule")  needs to be 2 levels deep., recurse, depth should be args?

IstioRegistry should resolve it to a key and call down into ConfigRegistry.
Or the config registry needs to know the details of individual message types, which seems incorrect.